core/domain/user_services.py

```diff
@@ -920,6 +932,51 @@ def has_fully_registered_account(user_id):
         feconf.REGISTRATION_PAGE_LAST_UPDATED_UTC)
 
 
+def get_all_profiles_auth_details_by_parent_user_id(parent_user_id):
+    """Gets domain objects representing the auth details for all profiles
+    associated with the user having the given parent_user_id.
+
+    Args:
+        parent_user_id: str. User id of the parent_user whose associated
+            profiles we are querying for.
+
+    Returns:
+        list(UserAuthDetails). The UserAuthDetails domain objects
+        corresponding to the profiles linked to given parent_user_id. If that
+        parent user does not have any profiles linked to it, the
+        returned list will be empty.
+    """
+    get_auth_details_by_user_id(parent_user_id, strict=True)
+    return [
+        _get_user_auth_details_from_model(model) for model in
+        user_models.UserAuthDetailsModel.get_all_profiles_by_parent_user_id(
+            parent_user_id)
+    ]
+
+
+def does_profile_with_display_alias_exist(parent_user_id, display_alias):
+    """Checks whether a given display alias value already exists for some
+        user within the account identified by the given parent_user_id.
+
+    Args:
+        parent_user_id: str. User id of the parent (full) user in the account.
+        display_alias: str. The queried display alias value to be checked.
+
+    Returns:
+        bool. Whether the display alias is unique within the account among all
```

Comment:
This still uses the word "unique" and is therefore not clear.

Reply:
Sorry but I am not sure what else to write - this is the exact thing I want to do i.e check whether the given display alias is unique in the account among all user profiles. Why does word "unique" not sound clear to you? I am able to understand currently.

Reply:
The boolean actually denotes the existence of a record with the same display alias and parent id in the store.  'Unique in the account' is a bit vague since a profile is also a user (and a user can technically be an account too). You can reword this to something like 'Whether a given display alias value already exists for some user with the same parent user id'
Remember that other people not so familiar with this project may also be reading the documentation and code, so its important to remove any sort of ambiguity in the comments.

Reply:
Not using this function now.